File: Hypothesis_Assignment_Fantaloons.R

###################Proportional T Test(Fantaloons)##########

library(readxl)
Fantaloons <- read_excel(file.choose())
View(Fantaloons)
attach(Fantaloons)

#B.P:proportions of male and female walking in to the store is same or not

table1 <- table(Weekdays)
table1
table2 <- table(Weekend)
table2
table3 <- table(Weekdays,Weekend)
table3

#Ho= Proportions of Male and Female are same
#Ha= Proportions of Male and Female are not same
prop.test(x=c(47,167),c(113,287),conf.level = 0.95,alternative = "two.sided")
# two. sided -> means checking for equal proportions of Male and Female walking into stroe or not
# p-value = 0.003919 < 0.05 accept alternate hypothesis i.e. Unequal proportions


#find out whose proportion is higher
#Ho= Proportions of Male is less than Female
#Ha= Proportions of Male is greater than Female
prop.test(x=c(47,167),c(113,287),conf.level = 0.95,alternative = "greater")
# p-value = 0.998 > 0.05 accept null hypothesis 

#Proportions of Male is walking into store less than Female

File: Hypothesis_Assignment_Fantaloons.py
# -*- coding: utf-8 -*-
"""
Created on Sat Apr 18 19:51:15 2020

@author: RAVI
"""

import pandas as pd

from scipy import stats

import numpy as np

Fantaloons=pd.read_excel("C:\RAVI\Data science\Assignments\Modue 5 Hypothesis\Fantaloons.xlsx")
Fantaloons

from statsmodels.stats.proportion import proportions_ztest

tab1= Fantaloons.Weekdays.value_counts()
tab1
tab2=Fantaloons.Weekend.value_counts()
tab2

pd.crosstab(Fantaloons.Weekdays,Fantaloons.Weekend)

count = np.array([47,167])
nobs = np.array([113,287])

#Ho= Proportions of Male and Female are same
#Ha= Proportions of Male and Female are not same

stats,pval=proportions_ztest(count,nobs,alternative='two-sided')
print(pval)
# p-value = 0.0027 < 0.05 accept alternate hypothesis i.e. Unequal proportions

#find out whose proportion is higher
#Ho= Proportions of Male is less than Female
#Ha= Proportions of Male is greater than Female
stats,pval1=proportions_ztest(count,nobs,alternative='larger')
print(pval1)        
# p-value = 0.998 > 0.05 accept null hypothesis


#Proportions of Male is walking into store less than Female